locator/tablets.hh

```diff
@@ -228,6 +228,26 @@ struct resize_decision {
     sstring resize_name() const;
 };
 
+struct table_load_stats {
+    size_t size_in_bytes = 0;
+    // Stores the minimum phase among all replicas, as coordinator wants to know if
+    // all replicas are split ready for the current phase.
+    api::timestamp_type split_ready_phase{};
+
+    table_load_stats& operator+=(const table_load_stats& s) noexcept;
+    friend table_load_stats operator+(table_load_stats a, const table_load_stats& b) {
+        return a += b;
+    }
+};
```

Comment:
If we're migrating some tablet replicas, we could be double-counting. So we should not count tablet replicas that are the destination of streaming. Comment should go somewhere else but I thought of it here.

Reply:
It's amortized with large tablet count, but it's problematic with tables that are small in size. So we indeed need to handle it.

Reply:
Actually, when we're transitioning to cleanup stage, tablet count hasn't changed yet (that's the last step in the execution). The double accounting means that table size is growing, in the example, from 10G up to 20G, and back to 10G on tablet cleanup. For example, if avg size is 5G, then during migration, what could happen is that we'd have an unnecessary split (which would be revoked later) or merge cancellation.

As for the threshold for cancellation, we still need to define boundaries. Maybe 10% within target tablet size.